Resfeber/View Controllers/ExploreViewController.swift

```diff
@@ -134,6 +134,46 @@ extension ExploreViewController: WaterfallLayoutDelegate {
     }
 }
 
+// MARK: - Collection View Delegate
+
+extension ExploreViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, contextMenuConfigurationForItemAt indexPath: IndexPath, point: CGPoint) -> UIContextMenuConfiguration? {
+        guard let cell = collectionView.cellForItem(at: indexPath) as? DestinationCell,
+              let destination = cell.destination else { return nil }
+        
+        return UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { suggestedActions in
+            
+            // Setup Favorite menu item
+            
+            let addToFavoritesText = destination.isFavorite ? "Remove Favorite" : "Favorite"
+            
+            let favoriteImage: UIImage?
+            if destination.isFavorite {
+                favoriteImage = UIImage(systemName: "heart.fill")?.withTintColor(.systemRed, renderingMode: .alwaysOriginal)
+            } else {
+                favoriteImage = UIImage(systemName: "heart")
+            }
+            
+            let addToFavorites = UIAction(title: addToFavoritesText, image: favoriteImage) { action in
+                destination.isFavorite.toggle()
+            }
+            
+            // Setup Itinerary menu item
+            
+            let addToItinerary: UIAction
+            if destination.isOnItinerary {
+                addToItinerary = UIAction(title: "Added to Itinerary", image: UIImage(systemName: "briefcase"), attributes: .disabled, handler: {_ in})
+            } else {
+                addToItinerary = UIAction(title: "Add to Itinerary", image: UIImage(systemName: "briefcase")) { action in
```

Comment:
You probably want the closure to be a weak reference `[weak self] action in`

Reply:
This closure does not need to capture self, but only because of the unusual way we currently have the mock data setup. The mock data is stored as a static variable and is effectively being used as a singleton. We realize this setup is not ideal and will soon be refactoring this implementation so that the data will be accessed via an instance variable on DestinationController. Once this refactoring is completed we will declare [weak self] in the closure's capture list.

Reply:
This code was removed in PR #25 due to a change in requirements initiated by the client.